test/test_simplendf.py

```diff
@@ -23,12 +23,64 @@ def test_simpleread(self):
         self.assertEqual(indf.label, 'Signal')
         self.assertEqual(indf.units, 'counts')
         self.assertEqual(indf.title, 'Test Data')
+        self.assertEqual(indf.xnumb, 1 )
+        self.assertEqual(indf.dim, [5,3])
+
+        # Xtension sutf
+        # xnumb (property), xname, xloc,  xstat, xdel,  xnew
+        self.assertEqual(indf.xname(0), 'FITS')
+        self.assertEqual(indf.xstat('FITS'), True)
+        self.assertEqual(indf.xstat('NONEXISTENT'), False)
+        self.assertIsInstance(type(indf.xloc('FITS', 'READ')), type(hds.HDSWrapperClass))
+
+
+        # AXSIS  stuff
+        # aform, astat, acget, anorm, amap,  aread,
+        self.assertEqual(indf.astat('LABEL', 0), True)
+        self.assertEqual(indf.astat('UNIT', 0), True)
+        self.assertEqual(indf.astat('WIDTH', 0), False)
+        self.assertEqual(indf.astat('CENTRE', 0), True)
+        self.assertEqual(indf.astat('VARIANCE', 0), False)
+
+        self.assertEqual(indf.astat('LABEL', 1), True)
+        self.assertEqual(indf.astat('UNIT', 1), True)
+        self.assertEqual(indf.astat('WIDTH', 1), False)
+        self.assertEqual(indf.astat('CENTRE', 1), True)
+        self.assertEqual(indf.astat('VARIANCE', 1), False)
+
+        self.assertEqual(indf.acget('LABEL', 1), 'Right ascension')
+        self.assertEqual(indf.acget('LABEL', 0), 'Declination')
+        self.assertEqual(indf.acget('UNIT', 1), 'deg')
+        self.assertEqual(indf.acget('UNIT', 0), 'deg')
+
+        # Main ndf stuff
+        history = indf.history()
+        self.assertEqual(len(history), 4)
+        self.assertEqual(history[0].application.split()[0], 'HISSET')
+        self.assertEqual(indf.type('DATA'), '_REAL')
+        self.assertSequenceEqual(indf.bound(), [[1,1], [5,3]])
+
+        data = indf.read('DATA')
+        data_shouldbe = (np.ones([5,3])*-3.4028235e+38).astype(np.float32)
+        self.assertSequenceEqual(data.tolist(), data_shouldbe.tolist())
+
+        #  cget, , new, map (and using it to change comp),
+        #  annul, , state
+        from starlink import Ast
+        self.assertIsInstance(indf.gtwcs(), Ast.FrameSet)
+
+        self.assertEqual(indf.state('LABEL'), True)
+        self.assertEqual(indf.state('QUALITY'), False)
+        with self.assertRaises(hds.StarlinkError):
+            indf.state('NONEXISTANT')
+
+        indf.annul()
```

Comment:
Reminds me that this really needs a context manager so that you can do
```python
with ndf.open() as indf:
    stuff
```
and have it automatically close the file.

Reply:
Oh, thats a really good idea.

Reply:
The word nonexistant is non-existent.  (-: